pinot-server/src/main/java/org/apache/pinot/server/api/resources/SegmentMetadataFetcher.java

```diff
@@ -108,48 +111,55 @@ private static Map<String, Map<String, String>> getIndexesForSegmentColumns(Segm
    * Helper to loop through column datasource to create a index map as follows for each column:
    * {<"bloom-filter", "YES">, <"dictionary", "NO">}
    */
-  private static Map<String, String> getColumnIndexes(DataSource dataSource) {
+  private static Map<String, String> getColumnIndexes(DataSource dataSource, ColumnMetadata columnMetadata) {
     Map<String, String> indexStatus = new LinkedHashMap<>();
     if (Objects.isNull(dataSource.getBloomFilter())) {
       indexStatus.put(BLOOM_FILTER, INDEX_NOT_AVAILABLE);
     } else {
-      indexStatus.put(BLOOM_FILTER, INDEX_AVAILABLE);
+      Long indexSize = columnMetadata.getIndexSizeMap().get(ColumnIndexType.BLOOM_FILTER);
+      indexStatus.put(BLOOM_FILTER, indexSize == null ? INDEX_AVAILABLE : String.valueOf(indexSize));
     }
 
     if (Objects.isNull(dataSource.getDictionary())) {
       indexStatus.put(DICTIONARY, INDEX_NOT_AVAILABLE);
     } else {
-      indexStatus.put(DICTIONARY, INDEX_AVAILABLE);
+      Long indexSize = columnMetadata.getIndexSizeMap().get(ColumnIndexType.DICTIONARY);
```

Comment:
> `indexSize == null ? INDEX_AVAILABLE`

Not sure I follow. Why put AVAILABLE when size is null ?

Reply:
AVAILABLE as it is configured but not yet generated (or the stats are not reflecting the current size).
which is the current behavior of the REST API

Reply:
we have to follow up with this in a separate PR. as this API is also used by the UI and it is currently not rendering the numbers correctly.

Reply:
hmm this seems confusing. I am ok with following up separately. Didn't get a chance to review the previous PR so not sure of the full context.

If the API is called after the segments have been loaded / mapped on the server, then all indexes should exist right.

So may be the response could be something like

- <some> INDEX available in all segments with avg / aggregated size <blah>
- <some> INDEX available on N segments with avg / aggregated size <blah>
- <some> INDEX not available / size unknown on M segments

I am guessing size not available will happen when server has not downloaded + loaded all segments and / or there is some bug ?

Reply:
ah I see the confusion. so together with #9712 .
we have 4 APIs touched 
- `/tables/{tableName}/segments/{segmentName}/metadata` (modified in #9712)
  - `/segments/{tableName}/metadata` is also affected
- `/tables/{tableName}/metadata` (current PR modified this)
- `/segments/{tableNameWithType}/{segmentName}` (this PR used to modify this but not anymore, b/c i found out that it is also used by the UI and it breaks the UI)